controller/users.go

```diff
@@ -118,7 +118,29 @@ func (c *UsersController) Create(ctx *app.CreateUsersContext) error {
 		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
 	}
 	if userExists {
-		return jsonapi.JSONErrorResponse(ctx, errors.NewVersionConflictError("user with such email or username already exists"))
+		// This may happen for manually deprovisioned users which are reactivating their account via the registration app
+		// We should re-deprovision such user
+		idn, err := c.app.UserService().IdentityByUsernameAndEmail(ctx, ctx.Payload.Data.Attributes.Username, ctx.Payload.Data.Attributes.Email)
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"err":      err,
+				"username": ctx.Payload.Data.Attributes.Username,
+				"email":    ctx.Payload.Data.Attributes.Email,
+			}, "unable to lookup identity by username and email")
+			return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
+		}
+		if idn == nil {
+			return jsonapi.JSONErrorResponse(ctx, errors.NewVersionConflictError("user with such email or username already exists"))
```

Comment:
Given that control entered this block because the call to `userExistsInDB(ctx context.Context, email string, username string) ` had returned a true, `idn==nil` is very unlikely to happen. If we retrieve only once ( as per my previous comment ), then we can get rid of this altogether.

Reply:
It can happen. If the reg app tries to create a user with already taken email (emails are not unique in oso) then we should return 409. We should deffer it from the case when we have a perfect match - username&email to some existing deprovisioned user.

Reply:
Understood!